app/com/gu/floodgate/contentsource/ContentSource.scala

```diff
@@ -16,13 +23,37 @@ case class ContentSourcesResponse(contentSources: Seq[ContentSource])
 case class SingleContentSourceResponse(contentSource: ContentSource)
 
 @JsonFormat
-case class ContentSource(id: String,
+case class ContentSource(
+    id: String,
     appName: String,
     description: String,
     reindexEndpoint: String,
-    environment: String) {
+    environment: String,
+    authType: String,
+    apiKey: Option[String]) {
 
   def uniqueId: String = s"$id-$environment"
+
+  def reindexUrl: String = {
+    val apiKeyParam = this.apiKey map (key => s"?api-key=$key") getOrElse ""
+    s"${this.reindexEndpoint}$apiKeyParam"
+  }
+
+  def reindexUrlWithDateParams(dateParameters: DateParameters): String = {
+
+    def apiKeyParamwithPrefix(prefix: String): Option[String] = this.apiKey map (key => s"${prefix}api-key=$key")
+
+    val apiKeyWithQuestionMarkPrefix = apiKeyParamwithPrefix("?").getOrElse("")
+    val apiKeyWithAmpersandPrefix = apiKeyParamwithPrefix("&").getOrElse("")
+
+    (dateParameters.from, dateParameters.to) match {
+      case (None, None) => s"${this.reindexEndpoint}?$apiKeyWithQuestionMarkPrefix"
+      case (Some(f), None) => s"${this.reindexEndpoint}?from=${f.toString}$apiKeyWithAmpersandPrefix"
+      case (None, Some(t)) => s"${this.reindexEndpoint}?to=${t.toString}$apiKeyWithAmpersandPrefix"
+      case (Some(f), Some(t)) => s"${this.reindexEndpoint}?from=${f.toString}&to=${t.toString}$apiKeyWithAmpersandPrefix"
+    }
+  }
```

Comment:
Why are you constructing the URL manually? You can pass the URL params separately from the main URL when you construct the request using Play WS.

If you really want to build it yourself, how about this?

```scala
val apiKey: Option[String] = ...
val from: Option[String] = ...
val to: Option[String] = ...

def param(key: String, value: Option[String]): Option[String] = value.map(v => s"$key=$v")
val queryString: String = Seq(param("api-key", apiKey), param("from", from), param("to", to)).flatten.mkString("&")
s"$reindexEndpoint?$queryString"
```